app/src/main/java/org/glucosio/android/activity/AddGlucoseActivity.java

```diff
@@ -119,7 +119,11 @@ public void onNothingChosen(View labelledSpinner, AdapterView<?> adapterView) {
         if (this.isEditing()) {
             setTitle(R.string.title_activity_add_glucose_edit);
             GlucoseReading readingToEdit = presenter.getGlucoseReadingById(this.getEditId());
-            readingTextView.setText(readingToEdit.getReading() + "");
+            if (presenter.getUnitMeasuerement().equals("mg/dL")) {
+                readingTextView.setText(String.valueOf(readingToEdit.getReading()));
+            } else {
+                readingTextView.setText(String.valueOf(presenter.convertToMmol(readingToEdit.getReading())));
```

Comment:
Small duplication, you should determine text to be displayed in `if\else` and just set text after.